src/Compilers/CSharp/Portable/Emitter/Model/SynthesizedPrivateImplementationDetailsStaticConstructor.cs

```diff
@@ -51,16 +52,23 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                 body.Add(payloadInitialization);
             }
 
-            // MVID = TypeOf(PrivateImplementationDetails).Module.ModuleVersionId;
+            // MVID = System.Reflection.IntrospectionExtensions.GetTypeInfo(TypeOf(PrivateImplementationDetails)).Module.ModuleVersionId;
 
             BoundStatement mvidInitialization =
                 factory.Assignment(
                     factory.ModuleVersionId(),
                     factory.Property(
                         factory.Property(
-                            factory.TypeOfPrivateImplementationDetails(),
+                            factory.Convert(
+                                factory.WellKnownType(WellKnownType.System_Type),
+                                factory.Call(
+                                    null,
+                                    (MethodSymbol)Binder.GetWellKnownTypeMember(compilationState.Compilation, WellKnownMember.System_Reflection_IntrospectionExtensions__GetTypeInfo, diagnostics, syntax: syntax),
```

Comment:
Unlike GetWellKnownType API, GetWellKnownTypeMember can return null, which, I suspect, won't be handled properly. SyntheticBoundNodeFactory should have a helper that simply takes WellKnownMember value instead of a method symbol and takes care of things.